nptdms/tdms.py

```diff
@@ -228,6 +232,39 @@ def _ipython_key_completions_(self):
         """
         return list(self._groups.keys())
 
+    def __eq__(self, other):
```

Comment:
I don't think it makes sense to implement `__eq__` for a `TdmsFile`. This would be a breaking change for anybody relying on the current reference equality behaviour, for example if they're storing TdmsFiles in a set. Comparing all the channels, groups and properties also seems like a fairly niche thing to want to do, and for some use-cases people might expect different behaviour, like only returning True if the TdmsFiles are reading the same on-disk file.

This should probably just be a utility function in the tests. If you have a use-case for this outside of the npTDMS unit tests, then possibly it could be added as a standalone function, eg. `nptdms.testing.files_equal`, but I'm not sure that's really worth doing as this seems like a niche use-case and it's fairly straightforward to implement it yourself outside of npTDMS.